diff --git a/src/seccomp_notify.c b/src/seccomp_notify.c
index bbc06dd..fa94546 100644
--- a/src/seccomp_notify.c
+++ b/src/seccomp_notify.c
@@ -20,7 +20,6 @@
 #ifdef USE_SECCOMP
 
 #include <sys/sysmacros.h>
-#include <linux/seccomp.h>
 #include <seccomp.h>
 
 #include "seccomp_notify.h"
diff --git a/src/seccomp_notify_plugin.h b/src/seccomp_notify_plugin.h
index ab590b1..5598f20 100644
--- a/src/seccomp_notify_plugin.h
+++ b/src/seccomp_notify_plugin.h
@@ -2,7 +2,7 @@
 
 #ifdef USE_SECCOMP
 
-#include <linux/seccomp.h>
+#include <seccomp.h>
 
 struct seccomp_notify_conf_s {
 	const char *runtime_root_path;
