diff --git a/src/seccomp_notify.c b/src/seccomp_notify.c
index f6c755d..e672df0 100644
--- a/src/seccomp_notify.c
+++ b/src/seccomp_notify.c
@@ -8,7 +8,6 @@
 #include <errno.h>
 #include <seccomp.h>
 #include <sys/ioctl.h>
-#include <linux/seccomp.h>
 #include <sys/sysmacros.h>
 #include <dlfcn.h>
 #include <sys/wait.h>
diff --git a/src/seccomp_notify_plugin.h b/src/seccomp_notify_plugin.h
index b8c660a..16e4d7b 100644
--- a/src/seccomp_notify_plugin.h
+++ b/src/seccomp_notify_plugin.h
@@ -1,6 +1,6 @@
 #ifndef SECCOMP_NOTIFY_PLUGIN_H
 
-#include <linux/seccomp.h>
+#include <seccomp.h>
 
 #ifdef USE_SECCOMP
 
